mathics/builtin/combinatorial.py

```diff
@@ -440,9 +470,9 @@ def apply_2(self, list, n, evaluation):
                 
             elif n_len == 3:
                 elem1 = n.leaves[0].get_int_value()
-                elem2 = n.leaves[1].get_int_value()
+                elem2 = n.leaves[1].get_int_value() if not n.leaves[1].has_form('DirectedInfinity', 1) else len(list.leaves) + 1
                 elem3 = n.leaves[2].get_int_value()
-                if elem1 is None or elem2 is None or elem3 is None :
+                if elem1 is None or elem2 is None or elem3 is None or elem1 < 0 or elem2 < 0:
```

Comment:
Yes, it does. In the above case, the range(min_n, max_n, step_n) returns an empty list [] so the final result will be an empty list {}.